bool ramp_imp

---

bool success
